File: Python/121_best_time_to_buy_and_sell_stock.py
class Solution:
    def maxProfit(self, prices):
        if not prices:
            return 0
        min_num, max_stock = prices[0], 0
        for i in range(1, len(prices)):
            if prices[i] < min_num:
                min_num = prices[i]
            else:
                max_stock = max(max_stock, prices[i] - min_num)

        return max_stock

File: leetcode/archived/golang/121_best_time_to_buy_and_sell_stock.go
package main

func maxProfit(prices []int) int {
	if len(prices) == 0 {
		return 0
	}
	buy_num := prices[0]
	max_profit := 0
	for i := 1; i < len(prices); i++ {
		if (prices[i] < buy_num) {
			buy_num = prices[i]
		} else {
			max_profit = max(max_profit, prices[i]-buy_num)
		}
	}
	return max_profit
}

func max(a int, b int) int {
	if a > b {
		return a
	}
	return b
}
